src/main/kotlin/xyz/bluspring/kilt/loader/remap/KiltRemapper.kt

```diff
@@ -617,72 +677,78 @@ object KiltRemapper {
             .invoke(null, gameDir, gameId, gameVersion) as Path
     }
 
-    private fun getMCGameFile(): File? {
+    private fun getMCGameFile(): Path? {
         if (!FabricLoader.getInstance().isDevelopmentEnvironment) {
             val commonJar = GameProviderHelper.getCommonGameJar()
 
             if (commonJar != null)
-                return commonJar.toFile()
+                return commonJar
 
             val sidedJar = GameProviderHelper.getEnvGameJar(FabricLoader.getInstance().environmentType)
 
             if (sidedJar != null)
-                return sidedJar.toFile()
+                return sidedJar
 
             // this gives the obfuscated JAR, we don't want that
             //val inputGameJar = FabricLoader.getInstance().objectShare.get("fabric-loader:inputGameJar")
             //if (inputGameJar is Path)
-                //return inputGameJar.toFile()
+            //return inputGameJar.toFile()
 
             // This is our best bet towards getting the Intermediary JAR.
-            val deobfJar = getDeobfJarDir(FabricLoader.getInstance().gameDir, "minecraft", KiltLoader.MC_VERSION.friendlyString)
-                .resolve("${FabricLoader.getInstance().environmentType.name.lowercase()}-${launcher.targetNamespace}.jar")
-                .toFile()
+            val deobfJar =
+                getDeobfJarDir(
+                    FabricLoader.getInstance().gameDir,
+                    "minecraft",
+                    KiltLoader.MC_VERSION.friendlyString
+                ) / "${FabricLoader.getInstance().environmentType.name.lowercase()}-${launcher.targetNamespace}.jar"
 
             if (deobfJar.exists())
                 return deobfJar
         } else {
             // TODO: is there a better way of doing this?
-            val possibleMcGameJar = FabricLauncherBase.getLauncher().classPath.firstOrNull { path ->
-                val str = path.absolutePathString()
-                str.contains("net") && str.contains("minecraft") && str.contains("-loom.mappings.") && str.contains("minecraft-merged-")
-            } ?: return null
+//            val possibleMcGameJar = launcher.classPath.firstOrNull { path ->
+//                val str = path.absolutePathString()
+//                str.contains("net") && str.contains("minecraft") && str.contains("-loom.mappings.") && str.contains("minecraft-merged-")
+//            }
+
+            val possibleMcGameJar = Minecraft::class.java.protectionDomain.codeSource.location.toURI().toPath()
```

Comment:
Does this reliably work in dev environments? And most importantly, does this work without statically initializing the Minecraft class?

Reply:
working. no